minimesos/src/test/java/com/containersol/minimesos/main/MainTest.java

```diff
@@ -0,0 +1,122 @@
+package com.containersol.minimesos.main;
+
+import com.containersol.minimesos.cluster.MesosCluster;
+import org.apache.commons.io.FileUtils;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.contrib.java.lang.system.SystemOutRule;
+
+import java.io.File;
+import java.io.IOException;
+
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.reset;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class MainTest {
+
+    @Rule
+    public final SystemOutRule systemOutRule = new SystemOutRule().enableLog();
+
+    private Main main;
+
+    private CommandInfo commandInfo;
+    private CommandDestroy commandDestroy;
+    private CommandState commandState;
+    private CommandInstall commandInstall;
+    private CommandUp commandUp;
+
+    @Before
+    public void before() {
+        commandUp = spy(new CommandUp());
+        reset(commandUp);
+        when(commandUp.execute()).thenReturn(mock(MesosCluster.class));
```

Comment:
Although I believe mocks are for introducing unit test for legacy code only, this line is wrong: `commandUp.execute()` is invoked and it's result in passed to `when()`. Had to be replaced by
`doReturn(mock(MesosCluster.class)).when(commandUp).execute();`